src/frontend/org/voltdb/importclient/kinesis/KinesisStreamImporter.java

```
@@ -0,0 +1,369 @@
+import com.amazonaws.services.kinesis.clientlibrary.types.ProcessRecordsInput;
+import com.amazonaws.services.kinesis.clientlibrary.types.ShutdownInput;
+import com.amazonaws.services.kinesis.clientlibrary.types.ShutdownReason;
+import com.amazonaws.services.kinesis.model.Record;
+
+/**
+ * Importer implementation for Kinesis Stream importer. one instance of this per stream, per shard, per app
+ */
+public class KinesisStreamImporter extends AbstractImporter {
+
+    private KinesisStreamImporterConfig m_config;
+    private AtomicLong m_submitCount = new AtomicLong(0);
+    private AtomicLong m_cbcnt = new AtomicLong(0);
+
+    private Worker m_worker;
+
+    public KinesisStreamImporter(KinesisStreamImporterConfig config) {
+        m_config = config;
+    }
+
+    @Override
+    public URI getResourceID() {
+        return m_config.getResourceID();
+    }
+
+    @Override
+    public void accept() {
+
+        info(null, "Starting data stream fetcher for " + m_config.getResourceID().toString());
+        try {
+            KinesisClientLibConfiguration kclConfig = new KinesisClientLibConfiguration(m_config.getAppName(),
+                    m_config.getStreamName(), credentials(), UUID.randomUUID().toString());
+
+            kclConfig.withRegionName(m_config.getRegion()).withMaxRecords((int) m_config.getMaxReadBatchSize())
+                    .withInitialPositionInStream(InitialPositionInStream.TRIM_HORIZON)
+                    .withIdleTimeBetweenReadsInMillis(m_config.getIdleTimeBetweenReads())
+                    .withTaskBackoffTimeMillis(m_config.getTaskBackoffTimeMillis()).withKinesisClientConfig(
+                            KinesisStreamImporterConfig.getClientConfigWithUserAgent(m_config.getAppName()));
+
+            m_worker = new Worker.Builder().recordProcessorFactory(new RecordProcessorFactory()).config(kclConfig)
+                    .build();
+
+            m_worker.run();
+
+        } catch (Exception e) {
```

AWS eats all exceptions but IAE and re-throw it as RT exception.